### ISSUE TITLE
Cadvisor and influxdb,What is the correct way to plot the CPU utilization per container?

### ISSUE THREAD

**Issue by huikang** [google/cadvisor#679]
On a Docker host, I have 4 containers running including cadvisor and the stats are imported to influxdb.

(Result 1: cpu utilization of all the containers) The query of "select cpu_cumulative_usage from stats limit 600" yields

![cpu_all](https://cloud.githubusercontent.com/assets/463631/7474941/a93afc38-f312-11e4-99c8-45ddeac67d37.png)
 

(Result 2: cpu utilization of one container) The query of
"select cpu_cumulative_usage from stats  where container_name='mysql' limit 600"
yields

![cpu_one_container](https://cloud.githubusercontent.com/assets/463631/7474961/c697fd76-f312-11e4-8ce9-342d149af2ab.png)

As you see, result 2 seems not consider the difference and divide the time.
So the plot of result 2 does not look the same as in the cadvisor UI.
I am wondering what is the correct way to plot the CPU utilization in influxdb
for individual containers. Thanks.

**Comment by vmarmol** [comment]
Hi @huikang I believe you want to do:

```
select derivative(cpu_cumulative_usage) from stats where container_name='mysql' limit 600 group by time(2s)
```

**Comment by huikang** [comment]
@vmarmol Thanks for your comment. The results look better and closer to the output of cadvisor UI.
![influx-db-output](https://cloud.githubusercontent.com/assets/463631/7494750/931a09d8-f3d7-11e4-9029-3782674e55ba.png)

However, the Y-axis is not in percentage. Instead, is it in CPU cycles? Do you have any idea about how to get the output graph with the Y axis as in % unit as in cadvisor UI? Thanks.

**Comment by vmarmol** [comment]
cAdvisor also outputs in number of cores. The raw data is in billionths of a core so you'll want to divide by 1,000,000,000 to get the result in whole cores. For percentage you'll want to divide over the number of cores on the machine (or do percentage of a core, which is what cAdvisor does.)

**Comment by huikang** [comment]
@vmarmol 
Here is some raw output from "select derivative(cpu_cumulative_usage) from stats 
where time > 1430846420s and time < 1430846488s and container_name='mysql' group by time(5s)"

1430846485000	1007085805.3288821
1430846480000	1379218024.677719
1430846475000	1437682783.6420352
1430846470000	1497197523.5402424

My machine has 32 cores.
For the first point "1430846485000	1007085805.3288821", the cpu utilization is 
1437682783 / 1,000,000,000 / 32 = 0.043, which is 4.3% of all the cores. Is my calculation correct? Thanks.

**Comment by vmarmol** [comment]
This seems to be mostly correct (the second value is the derivative I believe). This will be the percentage used over that 5s interval.

**Comment by huikang** [comment]
@vmarmol 
Yes, you are right; the second value is derivative. Thanks for your clarification.

**Comment by vmarmol** [comment]
Great! Will close this @huikang but feel free to open another issue if you run into any trouble.

**Comment by mboussaa** [comment]
@vmarmol How can we get CPU% per core ?

**Comment by mboussaa** [comment]
@vmarmol Another question : how can we get the Memory% per container ?

**Comment by mboussaa** [comment]
@vmarmol what the unit of measure for CPU utilization. you said that it "outputs in number of cores". But how can you define this measure ? thanks

**Comment by vmarmol** [comment]
The raw CPU metric is in total nanoseconds used since creation. For the UI
with typically take the delta over a time period and divide it by the
length of the time period. This gives us instantaneous usage in number of
cores.

**Comment by mboussaa** [comment]
Having one core for example, I got 0.6 cores usage of my CPU using cadvisor UI. That means 60% of CPU utilization? 
If I would have the same result programmatically in influxDB (instead of getting the total in nanoseconds). I mean the percentage per core or the total percentage of cpu utilization. How can I manage to get the same results as the UI ? 

We already discussed this issue here https://github.com/google/cadvisor/issues/740
